typescript/infra/src/tx/transaction-reader.ts

```diff
@@ -0,0 +1,489 @@
+import { Result } from '@ethersproject/abi';
+import { decodeMultiSendData } from '@safe-global/protocol-kit/dist/src/utils/index.js';
+import {
+  MetaTransactionData,
+  OperationType,
+} from '@safe-global/safe-core-sdk-types';
+import { BigNumber, ethers } from 'ethers';
+
+import {
+  AnnotatedEV5Transaction,
+  ChainMap,
+  ChainName,
+  CoreConfig,
+  DerivedIsmConfig,
+  EvmIsmReader,
+  InterchainAccount,
+  MultiProvider,
+  coreFactories,
+  interchainAccountFactories,
+  normalizeConfig,
+} from '@hyperlane-xyz/sdk';
+import {
+  addressToBytes32,
+  bytes32ToAddress,
+  deepEquals,
+  eqAddress,
+  retryAsync,
+} from '@hyperlane-xyz/utils';
+
+import {
+  icaOwnerChain,
+  icas,
+  safes,
+} from '../../config/environments/mainnet3/owners.js';
+import { DeployEnvironment } from '../config/environment.js';
+import { getSafeAndService } from '../utils/safe.js';
+
+export class TransactionReader {
```

Comment:
probably a nit but feel like the functionality here is more of a `SafeTxChecker` than just a reader, wdyt?

Reply:
it's not overly Safe specific, like the fetching of txs from the Safe API lives outside this class. Maybe `GovernTransactionReader`?